crates/apub/src/activities/mod.rs

```diff
@@ -109,8 +104,8 @@ pub(crate) async fn verify_mod_action(
   let mod_ = mod_id.dereference(context).await?;
 
   let is_mod_or_admin =
-    CommunityView::is_mod_or_admin(&mut context.pool(), mod_.id, community.id).await?;
-  if is_mod_or_admin {
+    CommunityView::is_mod_or_admin(&mut context.pool(), mod_.id, community.id).await;
+  if is_mod_or_admin.is_ok() {
     return Ok(());
   }
 
```

Comment:
Some of these changes need to be fixed to correctly handle errors other than reading `false` in the query. For example, change this function to:

```
  // mod action comes from the same instance as the community, so it was presumably done
  // by an instance admin.
  // TODO: federate instance admin status and check it here
  if mod_id.inner().domain() == community.actor_id.domain() {
    return Ok(());
  }

  let mod_ = mod_id.dereference(context).await?;

  CommunityView::is_mod_or_admin(&mut context.pool(), mod_.id, community.id).await
```

If you don't want to change `NotAModerator` to `NotAModOrAdmin`, do this:


```
.map_err(|mut err| {
  if err.error_type == LemmyErrorType::NotAModOrAdmin {
     err.error_type = LemmyErrorType::NotAModerator;
  }
  err
})
```

An alternative is to use `LemmyResult<LemmyResult<()>>` for the return types, which would mean the only big change is that forgetting to use the inner value will cause a warning. This result should be checked with something like `result??` or `result? == Ok(())`, not with `is_ok` because accidental use of `result.is_ok()` instead of `result?.is_ok()` would be hard to notice.

Reply:
I'll fix this one. But also the `?.then_some(()).ok_or(LEMMY_ERROR).` will convert any Ok(false) or diesel error into the lemmyerror, so we should be okay. 

I'll make sure I get all the tests passing.

Reply:
I'm finding some (like the blocks), should actually be testing the `not(exists(` case, since a missing row should pass the check.